components/mpas-ocean/src/shared/mpas_ocn_time_average_coupled.F

```diff
@@ -473,7 +490,52 @@ subroutine ocn_time_average_coupled_accumulate(statePool, forcingPool, timeLevel
            end do
            !$omp end do
            !$omp end parallel
-        endif
+        elseif (trim(config_glc_thermal_forcing_coupling_mode) == '3d') then
+           call mpas_pool_get_array(forcingPool, 'avgThermalForcingAtZLevels', avgThermalForcingAtZLevels)
+           call mpas_pool_get_array(forcingPool, 'avgThermalForcingAtZLevelsMask', avgThermalForcingAtZLevelsMask)
+           call mpas_pool_get_array(forcingPool, 'glcZLevels', glcZLevels)
+           call mpas_pool_get_subpool(statePool, 'tracers', tracersPool)
+           call mpas_pool_get_array(tracersPool, 'activeTracers', activeTracers, 2)
+           call mpas_pool_get_dimension(tracersPool, 'index_temperature', indexTemperature)
+           call mpas_pool_get_dimension(tracersPool, 'index_salinity', indexSalinity)
+           call mpas_pool_get_dimension(tracersPool, 'nGlcZLevels', nGlcZLevels)
```

Comment:
I'm surprised this works.  I think it should be from the forcing pool:
```suggestion
           call mpas_pool_get_dimension(forcingPool, 'nGlcZLevels', nGlcZLevels)
```

Reply:
My recollection is that all dimensions get added to all var_structs, so either should work, but I agree this is the more appropriate way to do it